arcaflow_plugin_opensearch/opensearch_plugin.py

```diff
@@ -9,6 +9,49 @@
 from opensearch_schema import ErrorOutput, SuccessOutput, StoreDocumentRequest
 
 
+def convert_to_supported_type(value) -> typing.Dict:
+    type_of_val = type(value)
+    if type_of_val == list:
+        new_list = []
+        for i in value:
+            new_list.append(convert_to_supported_type(i))
+        # A list needs to be of a consistent type or it will
+        # not be indexible into a system like Opensearch
+        return convert_to_homogenous_list(new_list)
+    elif type_of_val == dict:
+        result = {}
+        for k in value:
+            result[convert_to_supported_type(k)] = convert_to_supported_type(value[k])
+        return result
+    elif type_of_val in (float, int, str, bool):
+        return value
+    elif isinstance(type_of_val, type(None)):
+        return str("")
+    else:
+        print("Unknown type", type_of_val, "with val", str(value))
+        return str(value)
+
+
+def convert_to_homogenous_list(input_list: list):
+    # To make all types in list homogeneous, we upconvert them
+    # to the least commom type.
```

Comment:
Return type hint?  Also, replace "commom" with "common" (although, I'm not sure that I would consider `str` to be "the least common" type...).